main.tf

```diff
@@ -2,158 +2,140 @@ data "azurerm_resource_group" "main" {
+      key_data = var.public_ssh_key != null ? var.public_ssh_key : tls_private_key.main.public_key_openssh
     }
   }
 
-  dynamic "default_node_pool" {
-    for_each = var.enable_auto_scaling == true ? [] : ["default_node_pool_manually_scaled"]
-    content {
-      orchestrator_version   = var.orchestrator_version
-      name                   = var.agents_pool_name
-      node_count             = var.agents_count
-      vm_size                = var.agents_size
-      os_disk_size_gb        = var.os_disk_size_gb
-      vnet_subnet_id         = var.vnet_subnet_id
-      enable_auto_scaling    = var.enable_auto_scaling
-      max_count              = null
-      min_count              = null
-      enable_node_public_ip  = var.enable_node_public_ip
-      availability_zones     = var.agents_availability_zones
-      node_labels            = var.agents_labels
-      type                   = var.agents_type
-      tags                   = merge(var.tags, var.agents_tags)
-      max_pods               = var.agents_max_pods
-      enable_host_encryption = var.enable_host_encryption
-    }
-  }
-
-  dynamic "default_node_pool" {
-    for_each = var.enable_auto_scaling == true ? ["default_node_pool_auto_scaled"] : []
-    content {
-      orchestrator_version   = var.orchestrator_version
-      name                   = var.agents_pool_name
-      vm_size                = var.agents_size
-      os_disk_size_gb        = var.os_disk_size_gb
-      vnet_subnet_id         = var.vnet_subnet_id
-      enable_auto_scaling    = var.enable_auto_scaling
-      max_count              = var.agents_max_count
-      min_count              = var.agents_min_count
-      enable_node_public_ip  = var.enable_node_public_ip
-      availability_zones     = var.agents_availability_zones
-      node_labels            = var.agents_labels
-      type                   = var.agents_type
-      tags                   = merge(var.tags, var.agents_tags)
-      max_pods               = var.agents_max_pods
-      enable_host_encryption = var.enable_host_encryption
-    }
+  default_node_pool {
+    orchestrator_version   = var.orchestrator_version
+    name                   = var.default_node_pool_name
+    node_count             = var.node_count
+    vm_size                = var.vm_size
+    os_sku                 = var.os_sku
+    os_disk_type           = var.os_disk_type
+    os_disk_size_gb        = var.os_disk_size_gb
+    vnet_subnet_id         = var.vnet_subnet_id
+    enable_auto_scaling    = var.enable_auto_scaling
+    max_count              = var.enable_auto_scaling == true ? var.max_count : null
+    min_count              = var.enable_auto_scaling == true ? var.min_count : null
+    enable_node_public_ip  = var.enable_node_public_ip
+    zones                  = var.zones
+    node_labels            = var.node_labels
+    node_taints            = var.node_taints
```

Comment:
According to [terraform-azurerm-provider pr #8982](https://github.com/hashicorp/terraform-provider-azurerm/pull/8982), since the [v2.35.0](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/CHANGELOG-v2.md#2350-november-05-2020), the `node_taints` argument is no longer be configured.

Reply:
I will change this field.